crates/sui-core/src/checkpoints/mod.rs

```diff
@@ -317,24 +340,71 @@ impl CheckpointBuilder {
 
     async fn create_checkpoint(
         &self,
-        mut batch: DBBatch,
-        l: Vec<TransactionEffects>,
+        mut effects: Vec<TransactionEffects>,
         last_checkpoint_of_epoch: bool,
-    ) -> SuiResult<DBBatch> {
+    ) -> Option<(CheckpointSummary, CheckpointContents)> {
         let last_checkpoint = self.tables.checkpoint_summary.iter().skip_to_last().next();
+        let epoch_rolling_gas_cost_summary = Self::get_epoch_total_gas_cost(
+            last_checkpoint.as_ref().map(|(_, c)| c),
+            &effects,
+            self.state.epoch(),
+        );
+        if last_checkpoint_of_epoch {
+            if let Err(err) = self
+                .augment_epoch_last_checkpoint(&epoch_rolling_gas_cost_summary, &mut effects)
+                .await
+            {
+                error!(
+                    "Failed to augment the last checkpoint of the epoch: {:?}",
+                    err
+                );
+                // TODO: Is returning None the best we can do here?
```

Comment:
I recommend simply returning Error here, so that checkpoint creation mechanism has a chance to retry checkpoint creation.
Returning None on error will cause the validator to completely diverge in terms of checkpoint creation in case of the failure here(which can very easily happen due to connectivity issues and timing out in `create_advance_epoch_tx_cert`, for example)